File: benchmarks/list_index.js
const { performance } = require('perf_hooks');

const t0 = performance.now();
const xs = new Array(5000);
for (let k = 0; k < 5000; k++) {
    xs[k] = k;
}
let total = 0;
for (let rep = 0; rep < 2000; rep++) {
    for (let i = 0; i < xs.length; i++) {
        total += xs[i];
    }
}
console.log(total);
console.log(`bench_ns=${Math.round((performance.now() - t0) * 1e6)}`);

File: benchmarks/list_index.rs
use std::time::Instant;

fn main() {
    let t0 = Instant::now();
    let mut xs = Vec::with_capacity(5000);
    for k in 0..5000 {
        xs.push(k as i64);
    }
    let mut total: i64 = 0;
    for _ in 0..2000 {
        for i in 0..xs.len() {
            total += xs[i];
        }
    }
    println!("{}", total);
    println!("bench_ns={}", t0.elapsed().as_nanos());
}
